auth/oauth2.go

```diff
@@ -0,0 +1,116 @@
+// Copyright 2023 Contributors to the Veraison project.
+// SPDX-License-Identifier: Apache-2.0
+package auth
+
+import (
+	"context"
+	"errors"
+	"fmt"
+	"net/url"
+	"time"
+
+	"github.com/mitchellh/mapstructure"
+	"golang.org/x/oauth2"
+)
+
+type Oauth2Authenticator struct {
+	TokenURL     string
+	ClientID     string
+	ClientSecret string
+	Username     string
+	Password     string
+
+	Token *oauth2.Token
+}
+
+func (o *Oauth2Authenticator) Configure(cfg map[string]interface{}) error {
+	decoded := struct {
+		TokenURL     string                 `mapstructure:"token_url"`
+		ClientID     string                 `mapstructure:"client_id"`
+		ClientSecret string                 `mapstructure:"client_secret"`
+		Username     string                 `mapstructure:"username"`
+		Password     string                 `mapstructure:"password"`
+		Rest         map[string]interface{} `mapstructure:",remain"`
+	}{}
+
+	if err := mapstructure.Decode(cfg, &decoded); err != nil {
+		return err
+	}
+
+	o.ClientID = decoded.ClientID
+	o.ClientSecret = decoded.ClientSecret
+	o.TokenURL = decoded.TokenURL
+	o.Username = decoded.Username
+	o.Password = decoded.Password
```

Comment:
Is there a requirement on Password encoding, like minimum characters and specific characters?

Reply:
Correct -- this would be up to the particular OAuth2 service, we should not enforce anything here.